### ISSUE TITLE
Network managing in CentOS7+ and Fedora 22+ (IP range)

### ISSUE THREAD

**Issue by qooob** [webmin/webmin#238]
Jamie, Fedora 22 is managing a bit differently network interfaces.

For example, instead of setting up IP range, you can now do (This is what Cockpit does):

```
IPADDR1=1.2.3.4
PREFIX1=29
IPADDR2=1.2.3.5
PREFIX2=29
IPADDR3=1.2.3.6
PREFIX3=29
```

Webmin doesn't recognize this correctly yet.

Besides, `/etc/resolve.conf` is automatically generated based on `ifcfg` file (on `service network restart`). For example, you can now put `DNS1, DNS2, DNS3 ...` key in `ifcfg`'s and `/etc/resolve.conf` will be automatically regenerated with `nameserver` and `search` keys based on `ifcfg`.

**Comment by qooob** [comment]
It will not let me use other IPs on the interface but the primary one. I created manually range0, range1 files and Webmin sees them but Fedora ignores them. They are always under the tab `Activated at Boot`. When I try to bind Webmin to use any IP from the range file it gives an error:

`Failed to change address : The IP address 1.2.3.3 is not active on this system, and so cannot be used`.

My idea is ether let Webmin manage even 'inactive` (that is represented in `Activated on Boot` tab) configuration (optionally) or make a quick fix for this crucial part to WORK. 

Please do something according to this problem as quick as possible :)

**Comment by qooob** [comment]
It would be really fine for me now if I could tweak some Webmin config file that would return Network Information BASED on Activated on Boot info but not on Active Now. How do I do it. Otherwise I will not be able to configure new system using Webmin/Virtualmin and would have to do it other way and then import VS, which I don't wanna do.

**Comment by qooob** [comment]
I was trying to feed `&active_interfaces` with `&boot_interfaces` function data, but it's a bit different and doesn't work as expected..

**Comment by qooob** [comment]
..but the best to make Webmin understand proper logic:

```
IPADDR1=1.2.3.4
PREFIX1=29
IPADDR2=1.2.3.5
PREFIX2=29
IPADDR3=1.2.3.6
PREFIX3=29
```

Seems to be the best to do

**Comment by qooob** [comment]
Apparently the same in CentOS7:
http://www.unixmen.com/linux-basics-assign-multiple-ip-addresses-single-network-interface-card-centos-7/

**Comment by qooob** [comment]
Is this hard to make Webmin understand `IPADDR1, IPADDR2, IPADDR3, IPADDR4..`.. because it's purely unusable the way it is now. I also tried to convince Fedora to read `-range` config files, like it used to be in CentOS 6 but with no luck.

**Comment by jcameron** [comment]
So, are those IPADDR1, 2, etc.. lines used only for interfaces that define a range of addresses?

**Comment by qooob** [comment]
I placed your fixes to correspondent files, restarted Webmin, so far no changes.

Before, I just user `ifcfg-name0` and made basic config there + defined all other IPv6 address. Then I did put the IPv4 additional address to the `ifcfg-name0-range0`, `ifcfg-name0-range1` file and so on.

Now you have only `ifcfg-name0` and place extra IPv4 using this example:

```
IPADDR1=1.2.3.4
PREFIX1=29
IPADDR2=1.2.3.5
PREFIX2=29
IPADDR3=1.2.3.6
PREFIX3=29
```

**Comment by jcameron** [comment]
The change I submitted only effects the DNSn lines .. I haven't done anything with IPADDRn lines yet.

Are these used only when an interface has multiple addresses, as a replacement for the old ifcfg-eth0:x files?

**Comment by qooob** [comment]
Yes, exactly!!

**Comment by jcameron** [comment]
Ok, this may be a bit tricky to support - currently Webmin has a lot of hard-coded assumptions about there being a 1-1 mapping between ifcfg-* files and IP addresses.

**Comment by qooob** [comment]
Well, making it dependent? Like rhel7+ or f22+ and centos7+? What is the other way to make it work on our systems? I could configure it somehow else if it helped.. 

Besides a all other latest software understands it correctly. I haven't tried yet, but I bet Virtualmin will also complain about "non-existing" IP addresses.

What should I do?

**Comment by jcameron** [comment]
The work-around for existing systems is to put each IP in a separate ifcfg-eth0:X file - this is what Virtualmin / Webmin will create for new interfaces.

Will Fedora 22 recognize those though? I know that CentOS 7 does..

**Comment by qooob** [comment]
Hmm. Well, let me tell you this. Yes, it works! :))

I set them all one by one. Webmin is happy as any other app.

BUT one

There is problem with Cockpit. (Official Fedora Administration Panel)

When you go to network settings and try to edit IP, it's not problem it reads them properly from your suggested configuration (ifcfg-eth0:X). The problem happens when you click SAVE button, it removes all of those `ifcfg-eth0:X` files and does it it's own, new way.

I immuned those cfg files and now it can't destroy them.. This is what we have. 

I think I will submit this as an issue, unless RHEL decided to remove it, they should fix it!

Thank you, Jamie!
